### ISSUE TITLE
Model definition

### ISSUE THREAD

**Issue by joaogsleite** [JWebCoder/funfunzmc#48]
This new model definition is being created on the branch: [feature/new-model-config](/JWebCoder/funfunzmc/tree/feature/new-model-config)

```typescript
{
  name: string,
  visible: boolean,
  relations: [
    {
      type: '1:n' | 'n:1' | 'm:n', 
      relationalTable: string,
      foreignKey: string,
      remoteForeignKey: string,
      remoteTable: string,
    },
    ...
  ],
  roles: {
    create: [string],  // array of role names
    read: [string],
    write: [string],
    delete: [string],
  },
  columns: [
    {
      name: string,
      searchable: boolean,  // field used on search
      visible: {
        list: boolean,  // field is returned on list requests
        detail: boolean,  // field is returned on single get request
        relation: boolean,  // field is returned on relation requests
      },
      model: boolean | {
        isPk: boolean,  // if column is primary key
        type: string,  // sql column type
        allowNull: boolean,
        ...  // other sql column options
      },
      layout: {
        isTitle: boolean,
        label: string,
        listColumn: boolean | {
          ...  // props for the frontend column component
        },
        editField: boolean | {
          type: string,
          ...  // other props for the frontend input component 
        },
      },
    },
    ...
  ],
  ...
  layout: {
    label: string,
    listPage: boolean | {
      chips: [
        { 
          label: string, 
          columns: [ { name: string, label: { key: value, ... } }, ],
        },
      ],
      ... // props for frontend table component
    },
    searchField: boolean | {
      ... // props for frontend search component
    },
    createButton: boolean | {
      ...  // props for frontend create button component
    },
    editButton: boolean | {
      ...  // other props for frontend edit button component
    },
    editPage: boolean | {
      sections: [
        { id: string, label: string, accordion: true|false, ... },  
      ],
      ...  // other props for frontend edit page
    },
    deleteButton: boolean | {
      ... // props for frontend delete button component
    },
  },
  ...
}
```

**Comment by JWebCoder** [comment]
#56

**Comment by JWebCoder** [comment]
This is has been resolved in #56
